core/modules/port_out.cc

```diff
@@ -118,29 +127,7 @@ int PortOut::OnEvent(bess::Event e) {
   if (e != bess::Event::PreResume) {
     return -ENOTSUP;
   }
-
-  const std::vector<bool> &actives = active_workers();
-
-  // Reclaim queues from workers that have been detached from this PortOut since
-  // the last resume.
-  for (size_t i = 0; i < Worker::kMaxWorkers; i++) {
-    if (!actives[i]) {
-      if (worker_queues_[i] >= 0) {
```

Comment:
Probably you still need an array something like `worker_to_queue`, which maps `wid -> qid`. The reason why is `wid` itself is not a reliable queue selector to be used in the critical path, since it can be an arbitrary number. Suppose there are two queues (Q0, Q1) and two workers(W0, W2). You would want W2 to use Q1, but the new code will use Q0 (without a lock, since there are supposed to be enough queues) as `2 % 2 == 0`.

Reply:
Ah great catch! I updated the pre-resume hook to round-robin the queues across the active workers. We also keep a per-queue worker count now to decide when to lock.

The worker-to-queue mapping is blown away and recomputed on each resume. I figured this is fine, though.